include/boost/dll/detail/posix/shared_library_impl.hpp

```diff
@@ -105,18 +115,23 @@ class shared_library_impl {
         if (!!(mode & load_mode::append_decorations)) {
             mode &= ~load_mode::append_decorations;
 
-            boost::filesystem::path actual_path = (
-                std::strncmp(sl.filename().string().c_str(), "lib", 3)
-                ? (sl.has_parent_path() ? sl.parent_path() / L"lib" : L"lib").native() + sl.filename().native()
-                : sl
-            );
-            actual_path += suffix();
+            boost::filesystem::path actual_path = append_decorations(sl);
 
             handle_ = dlopen(actual_path.c_str(), static_cast<native_mode_t>(mode));
             if (handle_) {
                 boost::dll::detail::reset_dlerror();
                 return;
             }
+            boost::system::error_code prog_loc_err;
+            boost::filesystem::path loc = boost::dll::detail::program_location_impl(prog_loc_err);
+            if (boost::filesystem::exists(actual_path) && !boost::filesystem::equivalent(sl, loc, prog_loc_err)) {
+                // decorated path exists : current error is not a bad file descriptor and we are not trying to load the executable itself
+                ec = boost::system::error_code(
+                    boost::system::errc::bad_file_descriptor,
```

Comment:
According to the comment above errc::bad_file_descriptor is not the right error code

Reply:
Indeed, but I didn't find any other boost::system::errc:: value that could fit with the current error category.
So I stuck to the "bad_file_descriptor" as here it is the real dlerror() error message that should refine the error.
Do you know of another value in boost::system::errc:: that could best fit the error at this place ?

Reply:
How about `executable_format_error`?

Reply:
Fixed in latest commit